ducktape/services/service.py

```diff
@@ -173,12 +176,33 @@ def start_node(self, node):
         """Start service process(es) on the given node."""
         raise NotImplementedError("%s: subclasses must implement start_node." % self.who_am_i())
 
-    def wait(self):
+    def wait(self, timeout_sec=600):
         """Wait for the service to finish.
         This only makes sense for tasks with a fixed amount of work to do. For services that generate
         output, it is only guaranteed to be available after this call returns.
         """
-        pass
+        alive_nodes = [] # nodes still alive following completion of wait
```

Comment:
Hope this isn't too bikesheddy - I have a slight preference for calling it `unfinished_nodes` since we've pulled the logic up to `Service`. Minor point though